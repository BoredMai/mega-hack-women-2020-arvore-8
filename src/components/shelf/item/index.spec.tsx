import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Item from '.';

describe('<Item>', () => {
  const item = { icon: 'princess.png', label: 'Princesa', url: 'princesa' };
  const props = {
    item,
    isSelected: false,
    onSelect: jest.fn().mockName('onSelect'),
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Item {...props} />) as ShallowWrapper;
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('adds class when item is disabled', () => {
    wrapper.setProps({ item: { ...item, disabled: true } });
    expect(wrapper.find('.itemDisabled')).toHaveLength(1);
  });

  it('adds class when item is selected', () => {
    wrapper.setProps({ isSelected: true });
    expect(wrapper.find('.itemSelected')).toHaveLength(1);
  });

  it('calls onSelect on button click', () => {
    wrapper.find('.item').simulate('click');

    expect(props.onSelect).toHaveBeenCalledWith(item);
  });
});
