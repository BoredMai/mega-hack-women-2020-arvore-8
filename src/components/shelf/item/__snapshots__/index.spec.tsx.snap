// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item> renders properly 1`] = `
<div
  className="item"
  onClick={[Function]}
>
  <img
    id="user-choice-Princesa"
    src="src/assets/icons/princess.png"
  />
  <label
    htmlFor="user-choice-Princesa"
  >
    Princesa
  </label>
</div>
`;
